boa/explorer.py

```diff
@@ -1,14 +1,29 @@
-import json
 import time
 from typing import Optional
 
-import requests
+from boa.rpc import json
 
-SESSION = requests.Session()
+try:
+    from requests_cache import CachedSession, Response
+
+    def _filter_fn(res: Response) -> bool:
+        data = res.json()
+        return _is_success_response(data) and not _is_rate_limited(data)
+
+    SESSION = CachedSession("etherscan_cache", filter_fn=_filter_fn)
```

Comment:
I think it's better to go to the user's global cache. It will improve cache sharing between repos (and maybe help with weird debugging situations)

Reply:
Which global cache?

Reply:
like `~/.cache/titanoboa`